apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/backend/ir/BackendIrBuilder.kt

```diff
@@ -754,4 +756,15 @@ internal class BackendIrBuilder constructor(
       val condition: BackendIr.Condition,
       val selectionKeys: Set<SelectionKey>,
   )
+
+  private val FrontendIr.Selection.Field.normalizedTypeName: String
+    get() {
+      val isListType = if (this.type is FrontendIr.Type.NonNull) {
+        this.type.ofType is FrontendIr.Type.List
+      } else {
+        this.type is FrontendIr.Type.List
+      }
+      val isObjectType = this.selections.isNotEmpty()
```

Comment:
Do we need to restrict to object types? If we have a list of scalar, we will never use this anyway?

Reply:
This is a generic function used for any field as well as when we compose selection key. So we need  this logic to not touch list of scalar types otherwise the name of field and it's selection key will be off.

Reply:
Do we care about selection key for scalar types? It's not like we're going to generate a model for them...

Reply:
we care as we use it as indication to add override modifier. So if selection keys for this particular field has a  different key than we currently building for we add override modifier.

Reply:
Got it, thanks !